modules/codegen/src/smithy4s/codegen/internals/CodegenImpl.scala

```diff
@@ -24,13 +24,17 @@ import smithy4s.codegen.transformers._
 import software.amazon.smithy.model.Model
 import software.amazon.smithy.model.node.Node
 import software.amazon.smithy.model.shapes.ModelSerializer
+import software.amazon.smithy.openapi.OpenApiConfig
 
 import scala.jdk.CollectionConverters._
 import software.amazon.smithy.model.transform.ModelTransformer
 
 private[codegen] object CodegenImpl { self =>
 
   def generate(args: CodegenArgs): CodegenResult = {
+    val smithyBuild = args.smithyBuild
```

Comment:
I tried to also implement the existing fields we already had in `SmithyBuild` here (`imports`, `dependencies` and `repositories`) but ran into problems with it, probably because I don't actually understand how the `ModelLoader` works particularly well. 😅 Especially since the PR was big enough already, I figured I'd just sent one implementation that acts on nothing but the OpenApiConfig and one can always iterate further from there.

Reply:
that's entirely fine imho